snippets/product-thumbnail.liquid

```diff
@@ -20,56 +21,43 @@
+
+  assign desktop_columns = 1
+  if desktop_layout == 'columns'
+    assign desktop_columns = 2
+  endif
+-%}
+
+{%- capture sizes -%}
+  (min-width: {{ settings.page_width }}px) {{ settings.page_width | minus: 100 | times: media_width | divided_by: desktop_columns | round }}px, (min-width: 990px) calc({{ media_width | times: 100 | divided_by: desktop_columns }}vw - 10rem), (min-width: 750px) calc((100vw - 11.5rem) / 2), calc(100vw - 4rem)
+{%- endcapture -%}
+
 <noscript>
   {%- if media.media_type == 'video' or media.media_type == 'external_video' -%}
     <span class="product__media-icon motion-reduce quick-add-hidden">{% render 'icon-play' %}</span>
     <div class="product__media media gradient global-media-settings" style="padding-top: {{ 1 | divided_by: media.preview_image.aspect_ratio | times: 100 }}%;">
-      <img
-        srcset="{% if media.preview_image.width >= 493 %}{{ media.preview_image | image_url: width: 493 }} 493w,{% endif %}
-          {% if media.preview_image.width >= 600 %}{{ media.preview_image | image_url: width: 600 }} 600w,{% endif %}
-          {% if media.preview_image.width >= 713 %}{{ media.preview_image | image_url: width: 713 }} 713w,{% endif %}
-          {% if media.preview_image.width >= 823 %}{{ media.preview_image | image_url: width: 823 }} 823w,{% endif %}
-          {% if media.preview_image.width >= 990 %}{{ media.preview_image | image_url: width: 990 }} 990w,{% endif %}
-          {% if media.preview_image.width >= 1100 %}{{ media.preview_image | image_url: width: 1100 }} 1100w,{% endif %}
-          {% if media.preview_image.width >= 1206 %}{{ media.preview_image | image_url: width: 1206 }} 1206w,{% endif %}
-          {% if media.preview_image.width >= 1346 %}{{ media.preview_image | image_url: width: 1346 }} 1346w,{% endif %}
-          {% if media.preview_image.width >= 1426 %}{{ media.preview_image | image_url: width: 1426 }} 1426w,{% endif %}
-          {% if media.preview_image.width >= 1646 %}{{ media.preview_image | image_url: width: 1646 }} 1646w,{% endif %}
-          {% if media.preview_image.width >= 1946 %}{{ media.preview_image | image_url: width: 1946 }} 1946w,{% endif %}
-          {{ media.preview_image | image_url }} {{ media.preview_image.width }}w"
-        src="{{ media | image_url: width: 1946 }}"
-        sizes="(min-width: {{ settings.page_width }}px) {{ settings.page_width | minus: 100 | times: media_width | round }}px, (min-width: 990px) calc({{ media_width | times: 100 }}vw - 10rem), (min-width: 750px) calc((100vw - 11.5rem) / 2), calc(100vw - 4rem)"
-        {% unless lazy_load == false %}loading="lazy"{% endunless %}
-        width="973"
-        height="{{ 973 | divided_by: media.preview_image.aspect_ratio | ceil }}"
-        alt="{{ media.preview_image.alt | escape }}"
-      >
+      {{ media.preview_image | image_url: width: 1946 | image_tag:
+        loading: lazy,
+        sizes: sizes,
+        widths: '493, 600, 713, 823, 990, 110, 1206, 1346, 1426, 1646, 1946',
```

Comment:
And here I wonder if we should add a smaller size as well because when setting the `desktop media size` to small an image container can be ~`190px` so maybe just adding something around `200`ish pixels.

Reply:
![](https://screenshot.click/25-27-4af6t-hfccq.png)

Reply:
Ludo, for clarification, when `desktop media size` is set to `small`, since the image container can be roughly ~`190px` in width + in terms of performance, it'd make more sense to instead have a smaller `width` other than `493` so it will use a more appropriate / accurate width, correct? If so, I'm totally aligned with this!

Reply:
Yeah exactly 🤔 👍 I think having a one smaller size to account for this situation would be ideal so we cover all the scenarios properly 🙂

Reply:
For some reason I'm not able to get it to output the `493` size, it only renders at `600` and that's using small media size and 2 columns. Of course if I remove the `600` size and check again, it'll use `493`. But maybe you can provide me with some context as I feel I might be missing something 🤔 

Additionally, I did some performance tests by removing both the `493` and `600` sizes and replaced it simply with `240` and I couldn't find any discernible difference when viewing products with varying media sizes. 

Perhaps we can look into this separately and not scope it into this PR. If this is something deemed to be something we should fix and improve, I'm thinking that we can easily slot it into either of the following:

- https://github.com/Shopify/dawn/issues/1931
- https://github.com/Shopify/dawn/issues/1936